services/QuillLMS/app/models/unit.rb

```diff
@@ -73,6 +73,16 @@ def email_lesson_plan
     end
   end
 
+  def self.create_with_incremented_name(user_id:, name: )
+    unit = Unit.create(user_id: user_id, name: name)
+    return unit if unit.persisted?
+    (2..20).each do |counter|
+      unit = Unit.create(user_id: user_id, name: "#{name} #{counter}")
+      return unit if unit.persisted?
+    end
+    false
```

Comment:
Per your other comment, we could keep the retry logic (and the more user friendly names), and have a less user friendly name as the fallback, e.g.
```suggestion
    Unit.create(user_id: user_id, name: "#{name} #{SecureRandom.uuid}")
```